src/sinol_make/commands/run/__init__.py

```diff
@@ -368,48 +377,98 @@ def print_view(output_file=None):
+					exit(1)
 
-			validator_program = self.config["subtasks"][subtask]["validator"].split()[0]
-			programs.append(validator_program)
-		programs = list(set(programs))
+				program = self.config["sinol_expected_scores"][prog]["program"].split()[0]
+				programs.append(program)
+			programs = list(set(programs))
 		compilation_results = self.compile_programs(programs)
 		os.makedirs(self.EXECUTIONS_DIR, exist_ok=True)
 		compiled_commands = []
-		for subtask in self.config["subtasks"]:
-			validator_program = os.path.join(self.EXECUTABLES_DIR, self.config["subtasks"][subtask]["validator"])
-			compiled_commands.append((subtask, validator_program, True))
-		names = list(self.config["subtasks"])
-		results = self.perform_executions(compiled_commands, names, programs, self.args.subtask_report)
-		for subtask in self.config["subtasks"]:
-			passed_groups = []
-			for group in self.config["scores"]:
-				if results[subtask][group] == "OK":
-					passed_groups.append(group)
-			passed_groups.sort()
-			should_pass = self.config["subtasks"][subtask]["groups"]
-			should_pass.sort()
-			if passed_groups != should_pass:
-				print(util.bold(util.color_red('Subtask %s will pass groups %s (expected %s).' % (subtask, passed_groups, should_pass))))
-				exit(1)
+		if suggestions:
+			for program in programs:
+				path = os.path.join(self.EXECUTABLES_DIR, program)
+				compiled_commands.append((program, path, True))
+			names = programs
+			results = self.perform_executions(compiled_commands, names, programs, self.args.expected_scores_report)
+
+			print(util.bold("Suggested expected scores description:"))
+			print("sinol_expected_scores:")
+
+			i = 1
+			new_config = self.config
+			new_config["sinol_expected_scores"] = {}
+
+			for program in programs:
+				print("  %s:" % program)
+				print("    program: %s" % program)
+				print("    expected: {" + ', '.join(results[program].items()) + "}")
+				new_config["sinol_expected_scores"][program] = {
+					"program": program,
+					"expected": results[program],
+					"points": 0
+				}
+
+				points = 0
+				for group, result in results[program].items():
+					if result == "OK":
+						points += self.scores[group]
+				print("    points: %d" % points)
+				new_config["sinol_expected_scores"][program]["points"] = points
+
+			if self.args.apply_suggestions:
+				print("Do you want to apply these suggestions? (y/n) ", end="")
+				choice = input().lower()
+				if choice == "y" or choice == "yes" or choice == "":
+					with open(os.path.join(os.getcwd(), "config.yml"), "w") as f:
+						yaml.dump(new_config, f, default_flow_style=False)
+					print(util.color_green("Suggestions applied."))
+				else:
+					print("Suggestions not applied.")
+		else:
+			for prog in self.config["sinol_expected_scores"]:
+				program = os.path.join(self.EXECUTABLES_DIR, self.config["sinol_expected_scores"][prog]["validator"])
```

Comment:
I thought we ditched the idea of validators in `sinol_expected_scores`? What is the definition of a validator in this context and what should it do?

Reply:
Yes, I must have missed this while refactoring

Reply:
It's a good habit to read your own PR from the GitHub interface -- with this method, you probably could catch most of the comments I wrote here, which saves time for both of us.

Reply:
OK, I’ll do it next time